mantidproject/mantid
# Update GetEiMonDet to version 3

**Description of work.**

This PR removes the heuristics relying on the nominal incident energy (as determined by the instrument's electronics/control software) and the elastic peak position tables which were used in `GetEiMonDet-v2`. Since the functionality and the algorithm's properties were changed, the version number was bumped to 3. Version 2 was marked as deprecated.

**Report to:** Björn Fåk, ILL 

**To test:**

Download the test file: 
[data.zip](https://github.com/mantidproject/mantid/files/2351597/data.zip)

The sample logs in the test file say the wavelength is 1.5Å. However, the instrument scientists say it should be around 1.3Å. For some reason the instrument control software has written a wrong value in the NeXus file.

Run the following script (fixing the path to the file):
```python
ws = LoadILLTOF('000001.nxs')
print('wavelength from sample logs: {}'.format(ws.run().getProperty('wavelength').value))
epps = FindEPP(ws)
pulseInterval = ws.run().getProperty('pulse_interval').value
E_i2 = GetEiMonDet(DetectorWorkspace=ws, IndexType='Workspace Index', 
    Detectors='0-299', DetectorEPPTable=epps, Monitor=396, PulseInterval=pulseInterval,
    Version=2)
lambda2 = UnitConversion.run('Energy', 'Wavelength', E_i2, ws.spectrumInfo().l1(), 2.0, 0., DeltaEModeType.Direct, 0)
print('wavelength from algorithm version 2: {}'.format(lambda2))
E_i3 = GetEiMonDet(DetectorWorkspace=ws, DetectorWorkspaceIndexSet='0-299',
    MonitorIndex=396)
lambda3 = UnitConversion.run('Energy', 'Wavelength', E_i3, ws.spectrumInfo().l1(), 2.0, 0., DeltaEModeType.Direct, 0)
print('wavelength from algorithm version 3: {}'.format(lambda3))
```

Observe, that version 2 of `GetEiMonDet` is fooled by the sample logs to calibrate the incident energy to corresponding wavelength 1.5Å while version 3 produces the expected result.


Fixes #22833.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
